kernel/src/process/process_table.rs

```diff
@@ -14,43 +14,63 @@ use crate::{
     prelude::*,
 };
 
-static PROCESS_TABLE: Mutex<BTreeMap<Pid, Arc<Process>>> = Mutex::new(BTreeMap::new());
+static PROCESS_TABLE: Mutex<ProcessTable> = Mutex::new(ProcessTable::new());
 static PROCESS_GROUP_TABLE: Mutex<BTreeMap<Pgid, Arc<ProcessGroup>>> = Mutex::new(BTreeMap::new());
-static PROCESS_TABLE_SUBJECT: Subject<PidEvent> = Subject::new();
 static SESSION_TABLE: Mutex<BTreeMap<Sid, Arc<Session>>> = Mutex::new(BTreeMap::new());
 
 // ************ Process *************
-
 /// Gets a process with pid
 pub fn get_process(pid: Pid) -> Option<Arc<Process>> {
-    PROCESS_TABLE.lock().get(&pid).cloned()
+    PROCESS_TABLE.lock().get(pid).cloned()
 }
 
-pub(super) fn process_table_mut() -> MutexGuard<'static, BTreeMap<Pid, Arc<Process>>> {
+pub fn process_table_exclusive() -> MutexGuard<'static, ProcessTable> {
```

Comment:
I think using `process_table_mut` is better here to keep consistent with other tables. We currently use `session_table_mut`, `group_table_mut`, thus process table is better to use the same name.